src/main/java/core/gui/comp/table/HOTableModel.java

```diff
@@ -43,6 +51,15 @@ public abstract class HOTableModel extends AbstractTableModel {
 	/** instance of the same class **/
 	protected int instance;
 
+	public TableRowSorter<HOTableModel> getRowSorter() {
+		if ( table != null) return (TableRowSorter<HOTableModel>) table.getRowSorter();
+		if ( fixedColumnsTable != null) return fixedColumnsTable.getTableRowSorter();
+		return null;
+	}
+
+	private JTable table;
+	private FixedColumnsTable fixedColumnsTable;
```

Comment:
This feels like a kludge to have the two fields here: if I understand correctly, this could be one or the other.

I wonder if this could be solved by getting `initTable(FixedColumnsTable table)` to call `initTable(JTable table)` with the nested table in `FixedColumnsTable`? and then in `getRowSorter` get the sorter from `table` as normal.  Would that work?

Reply:
Thank you for your review - I'm also not happy about this switching of table types. I will check if I could handle it, the way you are suggesting,  making the FixedColumnClass an extension of Jtable.

Reply:
FixedColumnsTable is alone now.